src/main/scala/com/redislabs/provider/redis/rdd/RedisRDD.scala

```diff
@@ -38,11 +40,20 @@ class RedisKVRDD(prev: RDD[String],
   def getKV(nodes: Array[RedisNode], keys: Iterator[String]): Iterator[(String, String)] = {
     groupKeysByNode(nodes, keys).flatMap { case (node, nodeKeys) =>
       val conn = node.endpoint.connect()
-      val stringKeys = filterKeysByType(conn, nodeKeys, "string")
-      val response = mapWithPipeline(conn, stringKeys) { (pipeline, key) =>
+
+      val response = mapWithPipeline(conn, nodeKeys) { (pipeline, key) =>
         pipeline.get(key)
       }
-      val res = stringKeys.zip(response).iterator.asInstanceOf[Iterator[(String, String)]]
+
+      val res = nodeKeys.zip(response)
+        .view
```

Comment:
Thanks @mayankasthana for the changes!
One minor question - what is the reason we need `view` here?

Reply:
We don't explicity need it. We are doing two `flatMap` operations, so I think it is creating two full collections, and we don't really need the intermediate collection. I was trying to make it lazy.

Reply:
I see. Can we replace `.view` with `.iterator` and remove `.iterator` from the last line?

Reply:
Yes, we can do that. 
But now that I think about it, I think I changed the semantics. Earlier we used to give out an `Iterator` of a strict collection and I changed it to become lazy. I think doing `.force.iterator` or `.collect.iterator` at the end is better, so that we evaluate the whole collection before giving out the `Iterator`. This way, we also throw any exceptions immediately and not need to wait for `.next` to be called by the caller.

Reply:
I just removed the `view` thing to not make things complicated.